python/glow/gwas/logistic_regression.py

```diff
@@ -0,0 +1,185 @@
+        fit_intercept : Whether or not to add an intercept column to the covariate DataFrame
+        values_column : A column name or column expression to test with linear regression. If a column name is provided,
+                        `genotype_df` should have a column with this name and a numeric array type. If a column expression
+                        is provided, the expression should return a numeric array type.
+        dt : The numpy datatype to use in the linear regression test. Must be `np.float32` or `np.float64`.
+    '''
+
+    gwas_fx._check_spark_version(genotype_df.sql_ctx.sparkSession)
+    gwas_fx._validate_covariates_and_phenotypes(covariate_df, phenotype_df, is_binary=True)
+    sql_type = gwas_fx._regression_sql_type(dt)
+    genotype_df = gwas_fx._prepare_genotype_df(genotype_df, values_column, sql_type)
+    result_fields = [
+        # TODO: Probably want to put effect size and stderr here for approx-firth
+        StructField('tvalue', sql_type),
+        StructField('pvalue', sql_type),
+        StructField('phenotype', StringType())
+    ]
+
+    result_struct = gwas_fx._output_schema(genotype_df.schema.fields, result_fields)
+    C = covariate_df.to_numpy(dt, copy=True)
+    if fit_intercept:
+        C = gwas_fx._add_intercept(C, phenotype_df.shape[0])
+    Y = phenotype_df.to_numpy(dt, copy=True)
+    Y_mask = ~(np.isnan(Y))
+    np.nan_to_num(Y, copy=False)
+
+    state = gwas_fx._loco_make_state(
+        Y, phenotype_df, offset_df,
+        lambda y, pdf, odf: _assemble_log_reg_state(y, pdf, odf, C, Y_mask))
+
+    def map_func(pdf_iterator):
+        for pdf in pdf_iterator:
+            yield gwas_fx._loco_dispatch(pdf, state, _logistic_regression_inner, C,
+                                         Y_mask.astype(np.float64), fallback,
+                                         phenotype_df.columns.to_series().astype('str'))
+
+    return genotype_df.mapInPandas(map_func, result_struct)
+
+
+@typechecked
+def _logistic_null_model_predictions(
+    y: NDArray[Any, Float],
+    X: NDArray[Any, Float],
+    y_mask: NDArray[Any, bool],
+    # nptying can't handle optional NDArrays, so don't verify the type for offset
+    offset
+) -> NDArray[Any, Float]:
+    if offset is not None:
+        offset = offset[y_mask]
+    model = sm.GLM(y[y_mask],
+                   X[y_mask],
+                   family=sm.families.Binomial(),
+                   offset=offset,
+                   missing='ignore')
+    fit_result = model.fit()
+    predictions = model.predict(fit_result.params)
+
+    # Store 0 as prediction for samples with missing phenotypes
+    remapped_predictions = np.zeros(y.shape)
+    remapped_predictions[y_mask] = predictions
+    return remapped_predictions
+
+
+@dataclass
+class LogRegState:
+    inv_CtGammaC: NDArray[(Any, Any), Float]
+    gamma: NDArray[(Any, Any), Float]
+    Y_res: NDArray[(Any, Any), Float]
+
+
+@typechecked
+def _assemble_log_reg_state(
```

Comment:
nit: you use `create` for linear regression and `assemble` here. Perhaps better to use same verb in both.